server/controllers/todoControllers.js

```diff
@@ -0,0 +1,68 @@
+import asyncHandler from "express-async-handler"
+import { Todo } from "../models/todoModel.js"
+
+// @desc    Get all todos
+// @route   GET "/api/todo/"
+const getAllTodos = asyncHandler(async (req, res) => {
+    const getTodos = await Todo.find().sort({ createdAt : -1 })
+
+    if(getTodos){
+        res.status(201).json({message: "All todos fetched", data: getTodos})
+    }else{
+        res.status(500).json({error : "Server error"})
+    }
+})
+
+// @desc    Create a new todo
+// @route   POST "/api/todo/createTodo"
+const createTodo = asyncHandler(async (req, res) => {
+    const {todo, priority} = req.body
+
+    if(!todo){
+        res.status(400).json({error: "All fields are required!"})
+        return;
+    }
+
+    const newTodo = await Todo.create({
+        todo,
+        priority
+    })
+
+    if(newTodo){
+        res.status(201).json({message: "New todo created", data: newTodo})
+    }else{
+        res.status(500).json({error : "Server error"})
+    }
+})
+
+// @desc    Mark as complete
+// @route   PATCH "/api/todo/markCompleted/:id"
```

Comment:
the route should be api/todo/:id, in REST API patch request means its a update fuctionality